### PR TITLE
#162381200 Design the create article page

### PR DESCRIPTION
**Pull request by darthrighteous**
#### What does this PR do?
adds a new article page

#### Description of Task to be completed?
- add a create article page
- add state handlers for all inputs
- ensure category and tags fields autocompletes on user input
- ensure a user can add multiple tags
- ensure a user can add an image

#### How should this be manually tested?
- checkout to this branch
- run `npm install` to install necessary dependencies
- run `npm run dev`
- navigate to `localhost:8080/new-article`

#### Any background context you want to provide?
N/A

#### What are the relevant pivotal tracker stories?
[162381200](https://www.pivotaltracker.com/story/show/162381200)

#### Screenshots (if appropriate)
screen shot 2018-12-12 at 1 35 06 pm
screen shot 2018-12-12 at 1 38 58 pm

#### Questions:
N/A

#### Link to review app:
- [review app](https://valinor-ah-frontend-stag-pr-22.herokuapp.com/)

### CODE REVIEW

File: public/scripts/articlepage.js [hunk 1]
```diff
@@ -0,0 +1,2 @@
+const app = document.getElementById('article-category');
```
Comment (mbilesanmi) [inline public/scripts/articlepage.js]: What does this page do?

File: public/styles/index.scss [hunk 2]
```diff
@@ -8,6 +8,8 @@
 @import './Header';
 @import './Footer';
 @import './Navbar';
+@import './newarticle';
```
Comment (mbilesanmi) [inline public/styles/index.scss]: Why is this lowercase?

File: src/components/NewArticle/index.js [hunk 3]
```diff
@@ -0,0 +1,311 @@
...
+
+import { WithContext as ReactTags } from 'react-tag-input';
+import { sampleCategoryOptions, sampleTagOptions } from '../../../mockdata/samplebody';
+
+const KeyCodes = {
+  comma: 188,
+  enter: 13,
+  semicolon: 186
+};
+const delimiters = [KeyCodes.comma, KeyCodes.enter, KeyCodes.semicolon];
```
Comment (mbilesanmi) [inline src/components/NewArticle/index.js]: Would we not achieve the same with `Object.values(KeyCodes)`

File: src/components/NewArticle/index.js [hunk 4]
```diff
@@ -0,0 +1,311 @@
...
+                tagInputField: 'tag-input',
+                selected: 'tag-selection',
+                tag: 'tag-item',
+                remove: 'tag-remove',
+                suggestions: 'tag-menu-container',
+                activeSuggestion: 'tag-menu-item active'
+              }} />
+          </div>
+
+          <div className="save-draft-cell">
```
Comment (mbilesanmi) [inline src/components/NewArticle/index.js:297]: Why are we calling the btns cell?

File: src/index.js [hunk 5]
```diff
@@ -2,7 +2,6 @@ import React from 'react';
 import { render } from 'react-dom';
 import 'bootstrap/dist/css/bootstrap.min.css';
 import '../public/styles/index.scss';
-
```
Comment (mbilesanmi) [inline src/index.js]: Why remove this? If we remove this, add a line after the BS import.

**Review by mbilesanmi — CHANGES_REQUESTED** [review]
Finally, why is the review app broken?

File: src/components/NewArticle/index.js (hunk 3 above)
Comment (darthrighteous) [reply src/components/NewArticle/index.js]: Yes, changed this.

File: public/styles/index.scss (hunk 2 above)
Comment (darthrighteous) [reply public/styles/index.scss]: Fixed this

File: public/scripts/articlepage.js (hunk 1 above)
Comment (darthrighteous) [reply public/scripts/articlepage.js]: Nothing, shouldn't be here. Removed it

File: src/index.js (hunk 5 above)
Comment (darthrighteous) [reply src/index.js]: Fixed this

File: src/components/NewArticle/index.js (hunk 4 above)
Comment (darthrighteous) [reply src/components/NewArticle/index.js:297]: The way I designed the grid, the last two cells are for those buttons only

**Review by sulenchy — APPROVED** [review]
LGTM

**Review by tersoo-atsen — APPROVED** [review]

**Review by mbilesanmi — APPROVED** [review]

**Pull request merged**